File: src/main/java/com/atsuishio/superbwarfare/compat/thermoo/ThermooCompatHandler.java
package com.atsuishio.superbwarfare.compat.thermoo;

import com.atsuishio.superbwarfare.Mod;
import com.atsuishio.superbwarfare.compat.CompatHolder;
import com.atsuishio.superbwarfare.entity.vehicle.base.VehicleEntity;
import net.fabricmc.loader.api.FabricLoader;
import net.minecraft.server.level.ServerPlayer;

public class ThermooCompatHandler {
    private static final int VEHICLE_CORE_TEMPERATURE = 1;

    public static void init() {
        if (hasMod()) {
            Mod.LOGGER.info("Thermoo detected, enabling temperature compatibility for enclosed powered vehicles");
        }
    }

    public static void onPlayerInVehicle(ServerPlayer player) {
        if (!hasMod() || !(player.getVehicle() instanceof VehicleEntity vehicle)) {
            return;
        }

        if (vehicle.hasEnergyStorage()
                && vehicle.isEnclosed(vehicle.getSeatIndex(player))
                && vehicle.getEnergy() > 0
        ) {
            ThermooCompatApi.setTemperature(player, VEHICLE_CORE_TEMPERATURE);
        }
    }

    public static boolean hasMod() {
        return FabricLoader.getInstance().isModLoaded(CompatHolder.THERMOO);
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/compat/thermoo/ThermooCompatHandler.kt
package com.atsuishio.superbwarfare.compat.thermoo

import com.atsuishio.superbwarfare.Mod
import com.atsuishio.superbwarfare.compat.CompatHolder
import com.atsuishio.superbwarfare.entity.vehicle.base.VehicleEntity
import net.fabricmc.loader.api.FabricLoader
import net.minecraft.server.level.ServerPlayer

object ThermooCompatHandler {
    private const val VEHICLE_CORE_TEMPERATURE = 1

    @JvmStatic
    fun init() {
        if (hasMod()) {
            Mod.LOGGER.info("Thermoo detected, enabling temperature compatibility for enclosed powered vehicles")
        }
    }

    @JvmStatic
    fun onPlayerInVehicle(player: ServerPlayer) {
        if (!hasMod() || player.vehicle !is VehicleEntity) {
            return
        }

        val vehicle = player.vehicle as VehicleEntity

        if (vehicle.hasEnergyStorage()
            && vehicle.isEnclosed(vehicle.getSeatIndex(player))
            && vehicle.energy > 0
        ) {
            ThermooCompatApi.setTemperature(player, VEHICLE_CORE_TEMPERATURE)
        }
    }

    @JvmStatic
    fun hasMod(): Boolean {
        return FabricLoader.getInstance().isModLoaded(CompatHolder.THERMOO)
    }
}